### PR TITLE
[instant] Remove commonjs bundle, productionize UMD bundle

### PR DESCRIPTION
**Pull request by fragosti**
## Description

Should I change the CDN settings to redirect HTTP to HTTPS?

https://instant.0xproject.com/instant.js

## Checklist:

*   [ ] Prefix PR title with `[WIP]` if necessary.
*   [ ] Add tests to cover changes as needed.
*   [ ] Update documentation as needed.
*   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/instant/src/util/version.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const anyWindow = window as any;
+
+export const versionUtil = {
+    writeVersionInfoToWindow: () => {
+        anyWindow.__zeroExInstantGitSha = process.env.GIT_SHA;
```
Comment (steveklebanoff) [inline packages/instant/src/util/version.ts]: Can we write this on the `zeroExInstant` window object? I feel like that'd be much cleaner.

File: packages/instant/.production.discharge.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "domain": "instant.0xproject.com",
+    "build_command": "yarn build",
```
Comment (steveklebanoff) [inline packages/instant/.production.discharge.json]: I think we should change add  `--env.production` but I can do this when I update my heap PR ( see staging and dogfood env setting here: https://github.com/0xProject/0x-monorepo/pull/1295/files )

File: packages/instant/README.md [hunk 3]
```diff
@@ -2,35 +2,7 @@
...
-```json
-"compilerOptions": {
-    "typeRoots": ["node_modules/@0x/typescript-typings/types", "node_modules/@types"],
-}
-```
-
-**UMD Module**
-
-The package is also available as a UMD module named `zeroExInstant`.
+The package is available as a UMD module named `zeroExInstant`.
```
Comment (steveklebanoff) [inline packages/instant/README.md]: Time to add the production URL in the README here?

File: packages/instant/package.json [hunk 4]
```diff
@@ -5,13 +5,10 @@
...
-    "main": "lib/index.js",
-    "types": "lib/index.d.ts",
+    "main": "umd/instant.js",
+    "private": true,
     "scripts": {
-        "build": "yarn build:all",
-        "build:all": "run-p build:umd:prod build:commonjs",
-        "build:umd:prod": "webpack --mode production",
-        "build:commonjs": "tsc -b",
+        "build": "webpack --mode production",
```
Comment (steveklebanoff) [inline packages/instant/package.json]: simpler build process 💯

File: packages/instant/src/util/version.ts (hunk 1 above)
Comment (fragosti) [reply packages/instant/src/util/version.ts]: This is a great idea

File: packages/instant/.production.discharge.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    "upload_directory": "umd",
+    "index_key": "instant.js",
+    "error_key": "404.html",
+    "trailing_slashes": true,
+    "cache": 3600,
+    "aws_profile": "default",
```
[the remark this answers is not in the document]
Comment (BMillman19) [reply packages/instant/.production.discharge.json]: ok just looked at the other discharge configs and it looks like we left them using `default`

we need to update the other ones with the correct build command

[the remark this answers is not in the document]
Comment (steveklebanoff) [reply packages/instant/.production.discharge.json]: the 0xproject change is in the rollbar PR which I haven't merged yet ( https://github.com/0xProject/0x-monorepo/pull/1295 )

File: packages/instant/src/index.umd.ts [hunk 6]
```diff
@@ -110,3 +110,7 @@ export const render = (config: ZeroExInstantConfig, selector: string = DEFAULT_Z
     };
     window.onpopstate = onPopStateHandler;
 };
+
+// Write version info to the exported object for debugging
+export const GitSha = process.env.GIT_SHA;
```
Comment (steveklebanoff) [inline packages/instant/src/index.umd.ts]: nit but I would expect these to be in all caps constant-case `GIT_SHA`, but there may be an accepted pattern for this that 0x uses elsewhere?

File: packages/instant/package.json [hunk 7]
```diff
@@ -5,13 +5,10 @@
...
         "build:ci": "yarn build",
         "watch_without_deps": "tsc -w",
```
Comment (steveklebanoff) [inline packages/instant/package.json]: Is this command relevant any more?

File: packages/instant/.production.discharge.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
+    "domain": "instant.0xproject.com",
+    "build_command": "yarn build",
+    "upload_directory": "umd",
+    "index_key": "instant.js",
+    "error_key": "404.html",
+    "trailing_slashes": true,
+    "cache": 3600,
```
Comment (steveklebanoff) [inline packages/instant/.production.discharge.json]: Do we feel comfortable with this cache setting? Is there a way we could kill the cache if we had an urgent bug?

**Comment by steveklebanoff** [comment]
Also, we should remove the comment about the common js bundle in `webpack.config.js`

File: packages/instant/.production.discharge.json (hunk 8 above)
Comment (fragosti) [reply packages/instant/.production.discharge.json]: discharge also does it when you deploy a new version

Comment (steveklebanoff) [reply packages/instant/.production.discharge.json]: ok, sweet

File: packages/instant/package.json [hunk 9]
```diff
@@ -23,6 +19,7 @@
         "clean": "shx rm -rf lib coverage scripts",
         "deploy_dogfood": "discharge deploy -c .dogfood.discharge.json",
         "deploy_staging": "discharge deploy -c .staging.discharge.json",
+        "deploy_live": "discharge deploy -c .production.discharge.json",
```
Comment (steveklebanoff) [inline packages/instant/package.json]: nit but I would have expected this to be called `deploy_production`

Comment (fragosti) [reply packages/instant/package.json]: yea... it's meant to match the website but on second thought that isn't a good enough reason.

**Comment by steveklebanoff** [comment]
@fragosti Have you tried setting the compilerOption `declaration` to `false` to exclude the building of `.d.ts`?

**Pull request merged**